components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHost.java

```diff
@@ -85,7 +85,7 @@ public synchronized void registerHandler(HttpHandlerRegistrationInfo registratio
 
         String path = registrationInfo.getUri().getPath();
         String methods = registrationInfo.getMethodRestrict();
-        boolean prefixMatch = registrationInfo.isMatchOnUriPrefix();
+        boolean prefixMatch = !registrationInfo.isMatchOnUriPrefix();
```

Comment:
i thought i reverted the source back before checking in. I'd done it because if i don't invert, it seems all rest UTs are broken which was leading RootHandler to be registered twice and tests were getting broken completely. ı inverted and that explains what i got BUILD SUCCESS  then i checked in..